model/instance/lifecycle/destroy.go

```diff
@@ -72,51 +73,38 @@ func DestroyWithoutHooks(domain string) error {
 	return couchdb.DeleteDoc(couchdb.GlobalDB, inst)
 }
 
-func deleteAccounts(inst *instance.Instance) {
-	var accounts []*couchdb.JSONDoc
-	if err := couchdb.GetAllDocs(inst, consts.Accounts, nil, &accounts); err != nil || len(accounts) == 0 {
-		return
+func deleteAccounts(inst *instance.Instance) error {
+	var accounts []*account.Account
+	if err := couchdb.GetAllDocs(inst, consts.Accounts, nil, &accounts); err != nil {
+		if couchdb.IsNoDatabaseError(err) {
+			return nil
+		}
+		return err
 	}
 
-	ds := realtime.GetHub().Subscriber(inst)
-	defer ds.Close()
-
-	accountsCount := 0
-	for _, account := range accounts {
-		account.Type = consts.Accounts
-		if err := couchdb.DeleteDoc(inst, account); err == nil {
-			accountsCount++
+	var toClean []account.CleanEntry
+	for _, acc := range accounts {
+		// Accounts that are not tied to a konnector must not be deleted, and
+		// the aggregator accounts in particular.
+		slug := acc.AccountType
+		if slug == "" {
+			continue
 		}
+		man, err := app.GetKonnectorBySlug(inst, slug)
+		if err != nil {
+			return err
+		}
+		entry := account.CleanEntry{
+			Account:          acc,
+			Trigger:          nil, // We don't care, the triggers will all be deleted a bit later
```

Comment:
The triggers are deleted by what ? Can you explicit this here ?

Reply:
This file is about the deletion of an instance. The stack is brutal and destroy a lot of the instance in bulk (like the triggers). But the accounts are an exception, as we need to clean them on remote systems. The triggers are deleted here: https://github.com/cozy/cozy-stack/blob/master/model/instance/lifecycle/destroy.go#L51-L60